statistics/selectivity_test.go

```diff
@@ -570,6 +570,63 @@ func BenchmarkSelectivity(b *testing.B) {
 	pprof.StopCPUProfile()
 }
 
+func (s *testStatsSuite) TestStatsVer2(c *C) {
+	defer cleanEnv(c, s.store, s.do)
+	testKit := testkit.NewTestKit(c, s.store)
+	testKit.MustExec("use test")
+	testKit.MustExec("set tidb_analyze_version=2")
+
+	testKit.MustExec("drop table if exists tint")
+	testKit.MustExec("create table tint(a int, b int, c int, index singular(a), index multi(b, c))")
+	testKit.MustExec("insert into tint values (1, 1, 1), (2, 2, 2), (3, 3, 3), (4, 4, 4), (5, 5, 5), (6, 6, 6), (7, 7, 7), (8, 8, 8)")
+	testKit.MustExec("analyze table tint with 2 topn, 3 buckets")
+
+	testKit.MustExec("drop table if exists tdouble")
+	testKit.MustExec("create table tdouble(a double, b double, c double, index singular(a), index multi(b, c))")
+	testKit.MustExec("insert into tdouble values (1, 1, 1), (2, 2, 2), (3, 3, 3), (4, 4, 4), (5, 5, 5), (6, 6, 6), (7, 7, 7), (8, 8, 8)")
+	testKit.MustExec("analyze table tdouble with 2 topn, 3 buckets")
+
+	testKit.MustExec("drop table if exists tdecimal")
+	testKit.MustExec("create table tdecimal(a decimal(40, 20), b decimal(40, 20), c decimal(40, 20), index singular(a), index multi(b, c))")
+	testKit.MustExec("insert into tdecimal values (1, 1, 1), (2, 2, 2), (3, 3, 3), (4, 4, 4), (5, 5, 5), (6, 6, 6), (7, 7, 7), (8, 8, 8)")
+	testKit.MustExec("analyze table tdecimal with 2 topn, 3 buckets")
+
+	testKit.MustExec("drop table if exists tstring")
+	testKit.MustExec("create table tstring(a varchar(64), b varchar(64), c varchar(64), index singular(a), index multi(b, c))")
+	testKit.MustExec("insert into tstring values ('1', '1', '1'), ('2', '2', '2'), ('3', '3', '3'), ('4', '4', '4'), ('5', '5', '5'), ('6', '6', '6'), ('7', '7', '7'), ('8', '8', '8')")
+	testKit.MustExec("analyze table tstring with 2 topn, 3 buckets")
+
+	// test with clustered index
+	testKit.MustExec("set @@tidb_enable_clustered_index = 1")
+	testKit.MustExec("drop table if exists ct1")
+	testKit.MustExec("create table ct1 (a int, pk varchar(10))")
```

Comment:
Does this table miss its clustered index?

Reply:
oh yes, my fault.